pkg/mic/mic.go

```diff
@@ -295,6 +318,47 @@ func (c *Client) Sync(exit <-chan struct{}) {
 	}
 }
 
+func (c *Client) getVMSSNodes(refs map[string]bool) (map[string][]*corev1.Node, error) {
+	ls := make(map[string][]*corev1.Node)
+	for ref := range refs {
+		n, err := c.NodeClient.Get(ref)
```

Comment:
Can't we already identity whether vm is VMA or VMSS by the time we get the node client ?

Reply:
If we can I don't know where.